src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java

```diff
@@ -0,0 +1,60 @@
+package com.taskviewer.api.web.rq;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public record RqTaskSearchCriteria(String username, String status, Integer priority) {
+
+	@Contract(value = " -> new", pure = true)
```

Comment:
@cichan02 what is it?

Reply:
Thanks to this annotation, we can define a contract that our methods must match. JetBrains, the brand behind [IntelliJ IDEA](https://www.baeldung.com/intellij-basics), introduced the annotation. It allows the text editor to directly troubleshoot potential problems with the calling methods in our code.

The @Contract annotation has two attributes: value and pure. The value attribute contains clauses describing the relationship between the methods' inputs and output. 

A contract is a set of causality clauses of the form: “A -> B”. This means that providing A to the method will always give B as a result. For instance, “_ -> null” means that the method returns null for any input value.

Reply:
In my case that means, that method returns new for any input